crates/sui-types/src/message_envelope.rs

```diff
@@ -0,0 +1,113 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
```

Comment:
General comment: 
- Should we add to the abstraction of Envelope one more marker, to signify if it is Verified or NotVerified. 
- In that world a Envelope<... , Unverified> implements verify(committee) and returns Result<Envelope<..., Verified>> but verify on a verified thing is a no-op (in fact we do not have to implement it).
- We can of course have a `forget_verification` to get from Verified to Unverified